flytekit/annotated/task.py

```diff
@@ -82,12 +85,41 @@ def _create_and_link_node(self, ctx: FlyteContext, *args, **kwargs):
 
         return create_task_output(node_outputs)
 
+    def _local_execute(self, ctx: FlyteContext, **kwargs) -> Union[Tuple[Promise], Promise, None]:
+        """
+        This code is used only in the case when we want to dispatch_execute with outputs from a previous node
+        For regular execution, dispatch_execute is invoked directly.
+        """
+        # Unwrap the kwargs values. After this, we essentially have a LiteralMap
+        for k, v in kwargs.items():
+            if isinstance(v, Promise):
+                if not v.is_ready:
+                    raise BrokenPipeError(
+                        f"Expected an actual value from the previous step, but received an incomplete promise {v}")
+                kwargs[k] = v.val
+
+        input_literal_map = _literal_models.LiteralMap(literals=kwargs)
+
+        outputs_literal_map = self.dispatch_execute(ctx, input_literal_map)
```

Comment:
So i know this was done in another PR, when you removed the stupid BaseExecutor thing, but it harkens back to this discussion here: https://lyft.slack.com/archives/GAG9AAS9K/p1600964346001300 - where we were discussing whether a) tasks knew how to execute themselves b) they relied on an executor to execute them.

I feel like we all seemed inclined at the time to go with the latter option - you kinda wrote something like it in the google doc https://docs.google.com/document/d/1nJprPSZtxbXQLoafD2xeN0JIueHnEb6TNIsJiXHF6rY/edit

We're sure we're okay with this change?  This is the right thing to do?